lib/core/filtering/simple_rule.rb

```diff
@@ -0,0 +1,64 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+module Core
+  module Filtering
+    class SimpleRule
+      POLICY = 'policy'
+      FIELD = 'field'
+      RULE = 'rule'
+      VALUE = 'value'
+      ID = 'id'
+
+      DEFAULT_RULE_ID = 'DEFAULT'
+      DEFAULT_RULE = SimpleRule.new(
+        {
+          'policy' => 'include',
+          'field' => '_',
+          'rule' => 'regex',
+          'value' => '.*',
+          'id' => SimpleRule::DEFAULT_RULE_ID
+        }
+      )
+
+      attr_reader :policy, :field, :rule, :value, :id
+
+      def initialize(rule_hash)
+        @policy = rule_hash.fetch(POLICY)
+        @field = rule_hash.fetch(FIELD)
+        @rule = rule_hash.fetch(RULE)
+        @value = rule_hash.fetch(VALUE)
+        @id = rule_hash.fetch(ID)
+      end
+
+      def match?(document)
+        return true if id == DEFAULT_RULE_ID
+        doc_value = document[field]
+        case rule
+        when 'equals'
+          doc_value == value
+        when 'starts_with'
```

Comment:
Aha so these are going to be lowercase and with an underscore? That's something I haven't deduced from the implementation document, so I guess simple rules should be fixed accordingly.

Reply:
Yeah that was something I probably could have covered in the implementation document and I didn't. I don't really care how we serialize them as long as we're consistent. CC @sphilipse who will also need to align. @mchernyavskaya I'm happy to take the approach from your PR, though I may make small changes to make sure the casing is consistent across all rules.

Reply:
I actually prefer the snake_case since what we have is essentially an enum. We could just pull main in and refactor it in your branch.